components/greaselion/browser/greaselion_download_service.cc

```diff
@@ -185,21 +201,54 @@ GreaselionDownloadService::GreaselionDownloadService(
     is_dev_mode_ = true;
     resource_dir_ = forced_local_path;
     LoadDirectlyFromResourcePath();
+    dev_mode_task_runner_ = base::ThreadPool::CreateSequencedTaskRunner(
+        {base::TaskPriority::BEST_EFFORT,
+         base::TaskShutdownBehavior::CONTINUE_ON_SHUTDOWN, base::MayBlock()});
     dev_mode_path_watcher_ = std::make_unique<base::FilePathWatcher>();
-    if (!dev_mode_path_watcher_->Watch(resource_dir_,
-        true /*recursive*/,
-        base::Bind(&GreaselionDownloadService::OnDevModeLocalFileChanged,
-            weak_factory_.GetWeakPtr()))) {
-      LOG(ERROR) << "Greaselion could not watch filesystem for changes"
-          << " at path " << resource_dir_.LossyDisplayName();
-    }
+
+    using Callback = base::RepeatingCallback<void(bool)>;
+    base::FilePathWatcher::Callback file_path_watcher_callback =
+        base::BindRepeating(
+            [](scoped_refptr<base::SequencedTaskRunner> main_sequence,
+               const Callback& callback, const base::FilePath&, bool error) {
+              main_sequence->PostTask(FROM_HERE,
+                                      base::BindOnce(callback, error));
+            },
+            base::SequencedTaskRunnerHandle::Get(),
+            base::Bind(&GreaselionDownloadService::OnDevModeLocalFileChanged,
+                       weak_factory_.GetWeakPtr()));
+
+    // Start the watcher on a background sequence, reporting all events back to
+    // this sequence. base::Unretained is safe because the watcher instance
+    // lives on the target sequence and will be destroyed there in a subsequent
+    // task.
+    dev_mode_task_runner_->PostTaskAndReplyWithResult(
+        FROM_HERE,
+        base::BindOnce(&StartFilePathWatcher,
+                       base::Unretained(dev_mode_path_watcher_.get()),
```

Comment:
Just curious: is it safe to have a raw pointer here? Sometimes it's hard for me to tell.

Reply:
I believe so because we test for null in `StartFilePathWatcher` and there don't seem to be any other paths that lead to a dangling pointer.